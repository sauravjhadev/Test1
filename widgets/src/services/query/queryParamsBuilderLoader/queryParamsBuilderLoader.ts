import * as fs from 'fs';

import WidgetFileType, { FileLoaderResponse } from '../../../types';
import { messages } from '../../../messages';

export default function queryParamsBuilderLoader(widgetDir: string): Promise<FileLoaderResponse> {
    return new Promise((resolve, reject) => {
        fs.readFile(
            `${widgetDir}/${WidgetFileType.QUERY_PARAMS_BUILDER}`,
            'utf8',
            (error: Error, data: string) => {
                if (!data || error) {
                    reject(messages.invalidQueryParamsBuilder());
                }

                resolve({
                    type: WidgetFileType.QUERY_PARAMS_BUILDER,
                    data,
                });
            },
        );
    });
}
