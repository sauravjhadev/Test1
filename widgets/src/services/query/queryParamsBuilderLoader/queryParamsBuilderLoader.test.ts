import { messages } from '../../../messages';

import queryParamsBuilderLoader from './queryParamsBuilderLoader';

describe('queryParamsBuilderLoader', () => {
    it('correctly loads queryParamsBuilder', (done) => {
        queryParamsBuilderLoader('src/services/__fixtures__').then((data) => {
            expect(data).toMatchSnapshot();
            done();
        });
    });

    describe('when it fails to load the file', () => {
        it('should reject with an error message', () => {
            const result = queryParamsBuilderLoader('src/services/__fixtures__/dummy_path');

            expect(result).rejects.toEqual(messages.invalidQueryParamsBuilder());
        });
    });
});
