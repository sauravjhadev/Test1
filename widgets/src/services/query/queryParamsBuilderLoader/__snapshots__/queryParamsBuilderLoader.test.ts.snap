// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`queryParamsBuilderLoader correctly loads queryParamsBuilder 1`] = `
Object {
  "data": "{
    \\"productIds\\": {
        \\"reads\\": \\"products.*.productId\\",
        \\"type\\": \\"Int!\\"
    },
    \\"productId\\": {
        \\"reads\\": \\"productId\\",
        \\"type\\": \\"Int!\\"
    },
    \\"name\\": {
        \\"reads\\": \\"name\\",
        \\"type\\": \\"String!\\"
    }
}
",
  "type": "queryParamsBuilder.json",
}
`;
